File: src/my_project_typescript/medium_problems_typescript/from51to100/compareVersionNumbers.ts
function compareVersion(version1: string, version2: string): number {

    const v1 = version1.split('.').map((str) => parseInt(str))
    const v2 = version2.split('.').map((str) => parseInt(str))

    while (v1.length !== v2.length){
        if (v1.length < v2.length){
            v1.push(0)
        }
        else {
            v2.push(0)
        }
    }

    for (let i = 0; i < v1.length; i++){
        if (v1[i] > v2[i]) return 1;
        if (v1[i] < v2[i]) return -1
    }
    return 0

};

File: src/my_project/medium_problems/from101to150/compare_version_numbers.py
from typing import List 

class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        
        n = 4
        v1 = [int(v) for v in version1.split('.')]
        v2 = [int(v) for v in version2.split('.')]
        while len(v1) < n:
            v1.append(0)
        while len(v2) < n:
            v2.append(0)
        for i in range(n):
            if v1[i] > v2[i]:
                return 1
            elif v1[i] < v2[i]:
                return -1
        return 0
        
